src/Admin/Controllers/ProvidersController.cs

```diff
@@ -75,9 +84,18 @@ public async Task<IActionResult> Create(CreateProviderModel model)
             return View(model);
         }
 
-        await _providerService.CreateAsync(model.OwnerEmail);
+        var provider = model.ToProvider();
+        switch (provider.Type)
+        {
+            case ProviderType.Msp:
+                await _createProviderCommand.CreateMspAsync(provider, model.OwnerEmail);
```

Comment:
You could technically missuse the command and create a provider here and vise versa. Not sure if the appropiate solution would be to lift this into the command and do the check there.

Reply:
Rui originally did something like this, where there was a single `CreateProvider` method with this switch statement inside it. I actually gave the opposite feedback because the interface is different depending on the `ProviderType`, so I thought it made sense to separate them. Either would probably work, but seeing as how you don't feel strongly about it I think we can just leave it as is.